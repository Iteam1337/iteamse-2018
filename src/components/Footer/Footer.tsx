import gql from 'graphql-tag'
import React from 'react'
import { Query } from 'react-apollo'
import styled from 'styled-components'
import { FooterQuery } from '../../../typings/iteamse'
import { OPERATIONS_PAGE_QUERY } from '../../pages/Ops'
import GridColumn from '../Grid/GridColumn'
import NativeLink from '../Link/NativeLink'
import StyledPrefetchLink from '../Link/StyledPrefetchLink'
import UnstyledList from '../List/UnstyledList'
import iconFb from './img/icon_fb.svg'
import iconInstagram from './img/icon_instagram.svg'
import iconLinkedin from './img/icon_linkedin.svg'
import iconMynewsdesk from './img/icon_mynewsdesk.svg'
import iconYoutube from './img/icon_youtube.svg'
import Location from './Location'

export const FOOTER_QUERY = gql`
  query Footer {
    addresses {
      address1
      city
      contactMail
      contactPhone
      orgNumber
      title
      zip
    }
  }
`

const Wrap = GridColumn.extend`
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 40px;
  padding-top: 40px;

  @media (min-width: 1025px) {
    grid-row-gap: 40px;
    padding-left: 0;
    padding-right: 0;
  }
`

const FooterSections = styled.div`
  display: grid;
  grid-gap: 30px;

  @media (min-width: 1025px) {
    grid-template-columns: repeat(2, 315px) 1fr;
  }

  /* IE 11 */
  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    width: 100%;
    max-width: 1024px;
    display: flex;
    justify-content: space-around;

    > * {
      flex: 1 0 25%;
    }
  }
`

const Title = styled.div`
  font-weight: 500;
  margin-bottom: 20px;
`

const SocialMedia = styled.div``
const SocialMediaItem = styled.li`
  align-items: center;
  display: flex;
  margin-bottom: 10px;
`
const SocialMediaIcon = styled.img`
  margin-right: 15px;
`

const NativeLinkUnderlined = NativeLink.extend`
  border-bottom: 2px solid;
  color: black;
  text-decoration: none;
`

const SupportSection = styled.div`
  /* IE 11 */
  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    display: flex;
    width: 100%;
  }
`

class FooterQueryComponent extends Query<FooterQuery> {}

const Footer = () => {
  return (
    <FooterQueryComponent query={FOOTER_QUERY}>
      {({ loading, data }) => {
        if (loading || !data) {
          return null
        }

        const { addresses } = data

        return (
          <Wrap>
            <FooterSections>
              {addresses.map(address => (
                <Location key={address.zip} address={address} />
              ))}

              <SocialMedia>
                <Title>Följ oss</Title>
                <UnstyledList>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconFb} alt="" />
                    <NativeLinkUnderlined href="https://www.facebook.com/iteam1337/">
                      Facebook
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconInstagram} alt="" />
                    <NativeLinkUnderlined href="https://instagram.com/iteam1337">
                      Instagram
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconLinkedin} alt="" />
                    <NativeLinkUnderlined href="https://www.linkedin.com/company/270569/">
                      LinkedIn
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconYoutube} alt="" />
                    <NativeLinkUnderlined href="https://www.youtube.com/channel/UCU2TdLZ7p0jEuuGUxaod4lw">
                      YouTube
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconMynewsdesk} alt="" />
                    <NativeLinkUnderlined href="http://www.mynewsdesk.com/se/iteam">
                      MyNewsDesk
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                </UnstyledList>
              </SocialMedia>
            </FooterSections>
            <SupportSection>
              <StyledPrefetchLink
                black={true}
                query={OPERATIONS_PAGE_QUERY}
                to="ops"
              >
                Behöver du supporthjälp?
              </StyledPrefetchLink>
            </SupportSection>
          </Wrap>
        )
      }}
    </FooterQueryComponent>
  )
}

export default Footer
