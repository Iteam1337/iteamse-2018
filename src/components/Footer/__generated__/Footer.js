

/* @flow */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: Footer
// ====================================================

export type Footer_addresses = {
  __typename: "Address",
  address1: string,
  city: string,
  contactMail: string,
  contactPhone: string,
  orgNumber: string,
  title: string,
  zip: string,
};

export type Footer = {
  addresses: Array<Footer_addresses>
};

//==============================================================
// START Enums and Input Objects
// All enums and input objects are included in every output file
// for now, but this will be changed soon.
// TODO: Link to issue to fix this.
//==============================================================

//==============================================================
// END Enums and Input Objects
//==============================================================