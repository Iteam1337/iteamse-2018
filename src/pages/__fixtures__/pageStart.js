export const pageStart = {
  __typename: 'PageStart',
  headerImage:
    '//images.ctfassets.net/rj4r6yfcesw5/6WZT7iYMZGkuEUIc4S0KgK/499955f736b51dfde591a5195803a445/start_optimized.jpg',
  headerText1: 'Vi digitaliserar företag och organisationer',
  headerText2: 'genom strategi, kod och kultur',
  headerTextBgColor: '#FF3B5C',
  codeMobileImage:
    '//images.ctfassets.net/rj4r6yfcesw5/2mxfqVCqiw4CQYguMSYwYU/b9494a7ffcc913a275984ba7db02a1bf/dator.png',
  codeText:
    'Systemutveckling, automatisering, artificiell intelligens, maskininlärning. Vi har alla redskap du behöver för att bygga vad du vill. Det du vill. Och vet du inte vad du behöver så hjälper vi dig med det med.',
  codeTitle: 'Kod',
  codeLogo:
    '//images.ctfassets.net/rj4r6yfcesw5/3qnZSfySW4cEGoUo6KCAae/8a656eba15e15b61dacc9fbea38aa4bd/logo_af.png',
  codeSlug: 'arbetsformedlingen',
  codeLinkText: 'Läs om hur vi hjälper Arbetsförmedlingen med kod',
  cultureMobileImage:
    '//images.ctfassets.net/rj4r6yfcesw5/2y9WBlATsssIcSQmuQ6Ky8/32e27eb88cb9d3063be20803442e3a02/chat.png',
  cultureText:
    'Ibland ser vi oss som kulturarbetare. Det räcker inte att bygga digitala tjänster åt sina användare. Det är först när hela din organisation tänker, andas och lever digitalt som ni digitaliseras. på riktigt. Vi rustar dig för framtiden helt enkelt.',
  cultureTitle: 'Kultur',
  cultureLogo:
    '//images.ctfassets.net/rj4r6yfcesw5/4D844WR1v20ICIOG4SK0y8/15a008277f5565d9f52fbd5312f6e121/logo_seb.png',
  cultureSlug: 'seb',
  cultureLinkText: 'Läs om hur vi hjälper SEB med kultur',
  strategyMobileImage:
    '//images.ctfassets.net/rj4r6yfcesw5/1oqMfeZBnuQAAGweIGCKEW/d34548ee301d4d3f315e2ee313089093/Iteam-03-e.png',
  strategyText:
    'Vi digitaliserar Sverige. Pompigt! Men det betyder att vi hjälper dig att leda digitaliseringen istället för att vara den som blir lämnad på efterkälken. Både på den marknad där du verkar, och de marknader du inte ens visste att du borde verka på.',
  strategyTitle: 'Strategi',
  strategyLogo:
    '//images.ctfassets.net/rj4r6yfcesw5/1mv5fxBGLmeyWom4OI2WsI/a4d79cae2076b43aea926db2f1244b52/logo_tsab.png',
  strategySlug: 'tsab',
  strategyLinkText: 'Läs om hur vi hjälper Taxi Stockholm med strategi',
}
