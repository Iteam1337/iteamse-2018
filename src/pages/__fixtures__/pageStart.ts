export const pageStart = {
  __typename: 'PageStart',
  codeLinkText: 'Läs om hur vi hjälper Arbetsförmedlingen med kod',
  codeLogo:
    '//images.ctfassets.net/rj4r6yfcesw5/3qnZSfySW4cEGoUo6KCAae/8a656eba15e15b61dacc9fbea38aa4bd/logo_af.png',
  codeMobileImage:
    '//images.ctfassets.net/rj4r6yfcesw5/2mxfqVCqiw4CQYguMSYwYU/b9494a7ffcc913a275984ba7db02a1bf/dator.png',
  codeSlug: 'arbetsformedlingen',
  codeText:
    'Systemutveckling, automatisering, artificiell intelligens, maskininlärning. Vi har alla redskap du behöver för att bygga vad du vill. Det du vill. Och vet du inte vad du behöver så hjälper vi dig med det med.',
  codeTitle: 'Kod',
  contactTitle: 'Hör av dig till oss!',
  cultureLinkText: 'Läs om hur vi hjälper SEB med kultur',
  cultureLogo:
    '//images.ctfassets.net/rj4r6yfcesw5/4D844WR1v20ICIOG4SK0y8/15a008277f5565d9f52fbd5312f6e121/logo_seb.png',
  cultureMobileImage:
    '//images.ctfassets.net/rj4r6yfcesw5/2y9WBlATsssIcSQmuQ6Ky8/32e27eb88cb9d3063be20803442e3a02/chat.png',
  cultureSlug: 'seb',
  cultureText:
    'Ibland ser vi oss som kulturarbetare. Det räcker inte att bygga digitala tjänster åt sina användare. Det är först när hela din organisation tänker, andas och lever digitalt som ni digitaliseras. på riktigt. Vi rustar dig för framtiden helt enkelt.',
  cultureTitle: 'Kultur',
  headerImage:
    '//images.ctfassets.net/rj4r6yfcesw5/6WZT7iYMZGkuEUIc4S0KgK/499955f736b51dfde591a5195803a445/start_optimized.jpg',
  headerText1: 'Vi digitaliserar företag och organisationer',
  headerText2: 'genom strategi, kod och kultur',
  headerTextBgColor: '#FF3B5C',
  strategyLinkText: 'Läs om hur vi hjälper Taxi Stockholm med strategi',
  strategyLogo:
    '//images.ctfassets.net/rj4r6yfcesw5/1mv5fxBGLmeyWom4OI2WsI/a4d79cae2076b43aea926db2f1244b52/logo_tsab.png',
  strategyMobileImage:
    '//images.ctfassets.net/rj4r6yfcesw5/1oqMfeZBnuQAAGweIGCKEW/d34548ee301d4d3f315e2ee313089093/Iteam-03-e.png',
  strategySlug: 'tsab',
  strategyText:
    'Vi digitaliserar Sverige. Pompigt! Men det betyder att vi hjälper dig att leda digitaliseringen istället för att vara den som blir lämnad på efterkälken. Både på den marknad där du verkar, och de marknader du inte ens visste att du borde verka på.',
  strategyTitle: 'Strategi',
  team: ['rln', 'msr'],
}
