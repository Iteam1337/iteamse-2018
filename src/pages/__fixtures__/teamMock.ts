import { TEAM_QUERY } from '../../components/Team/Team'
import { teamMembers } from './teamMembers'

export const TeamQueryMock = {
  request: {
    query: TEAM_QUERY,
  },
  result: {
    data: {
      teamMembers,
    },
  },
}
