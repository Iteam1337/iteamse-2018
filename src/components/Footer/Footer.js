// @flow

import React from 'react'
import * as Iteam from '../../typings/iteam.flow'
import * as IteamCMS from './__generated__/Footer'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import Location from './Location'
import GridRow from '../Grid/GridRow'
import styled from 'styled-components'

type Props = Iteam.ApolloBase<IteamCMS.Footer>

const FooterQuery = gql`
  query Footer {
    addresses {
      address1
      city
      contactMail
      contactPhone
      orgNumber
      title
      zip
    }
  }
`

const Wrap = GridRow.extend`
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 40px;
  padding-top: 40px;

  @media (min-width: 1024px) {
    padding-left: 0;
    padding-right: 0;
  }
`

const Locations = styled.div`
  display: grid;
  grid-gap: 30px;

  @media (min-width: 1024px) {
    grid-template-columns: repeat(2, 315px);
  }
`

const Footer = () => {
  return (
    <Query query={FooterQuery}>
      {({ loading, data: { addresses } }: Props) => {
        if (loading) {
          return null
        }

        return (
          <Wrap>
            <Locations>
              {addresses.map(address => (
                <Location key={address.zip} address={address} />
              ))}
            </Locations>
          </Wrap>
        )
      }}
    </Query>
  )
}

export default Footer
