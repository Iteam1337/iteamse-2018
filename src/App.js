// @flow

import 'what-input'
import * as React from 'react'
import Route from 'react-router-dom/Route'
import Switch from 'react-router-dom/Switch'
import { injectGlobal } from 'styled-components'

import Home from './pages/Home'
import About from './pages/About'
import HowWeWork from './pages/HowWeWork'
import Team from './pages/Team'
import Cases from './pages/Cases'
import Case from './pages/Case'
import Work from './pages/Work'
import OpenPosition from './pages/OpenPosition'
import Ops from './pages/Ops'
import Footer from './components/Footer/Footer'
import MobileMenu from './components/MobileMenu/MobileMenu'

injectGlobal`
  body {
    -webkit-font-smoothing: antialiased;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    margin: 0;
    padding: 0;
  }
`

const App = () => (
  <React.Fragment>
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/om-oss" component={About} />
      <Route path="/hur-vi-jobbar" component={HowWeWork} />
      <Route path="/teamet" component={Team} />
      <Route exact path="/case" component={Cases} />
      <Route path="/case/:slug" component={Case} />
      <Route exact path="/jobba-hos-oss" component={Work} />
      <Route path="/jobba-hos-oss/:id" component={OpenPosition} />
      <Route path="/ops" component={Ops} />
    </Switch>
    <MobileMenu />
    <Footer />
  </React.Fragment>
)

export default App
