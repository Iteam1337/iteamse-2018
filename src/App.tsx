import * as React from 'react'
import { Route, Switch } from 'react-router-dom'
import { createGlobalStyle } from './theme'

import Footer from './components/Footer/Footer'
import MobileMenu from './components/MobileMenu/MobileMenu'
import About from './pages/About'
import Case from './pages/Case'
import Cases from './pages/Cases'
import Home from './pages/Home'
import HowWeWork from './pages/HowWeWork'
import Offers from './pages/Offers'
import OpenPosition from './pages/OpenPosition'
import Ops from './pages/Ops'
import Team from './pages/Team'
import TeamMember from './pages/TeamMember'
import Work from './pages/Work'

const GlobalStyle = createGlobalStyle`
  body {
    -webkit-font-smoothing: antialiased;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    margin: 0;
    padding: 0;
  }
`

const App = () => (
  <>
    <GlobalStyle />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/om-oss" component={About} />
      <Route path="/hur-vi-jobbar" component={HowWeWork} />
      <Route exact path="/medarbetare" component={Team} />
      <Route path="/medarbetare/:shortName" component={TeamMember} />
      <Route exact path="/case" component={Cases} />
      <Route path="/case/:slug" component={Case} />
      <Route exact path="/jobba-hos-oss" component={Work} />
      <Route path="/jobba-hos-oss/:id" component={OpenPosition} />
      <Route path="/ops" component={Ops} />
      <Route path="/erbjudanden" component={Offers} />
    </Switch>
    <MobileMenu />
    <Footer />
  </>
)

export default App
