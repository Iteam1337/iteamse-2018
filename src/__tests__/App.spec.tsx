import React from 'react'
import { render, wait } from 'react-testing-library'
import App from '../App'
import { addresses } from '../components/Footer/__fixtures__/addresses'
import { FOOTER_QUERY } from '../components/Footer/Footer'
import { pageStart } from '../pages/__fixtures__/pageStart'
import { TeamQueryMock } from '../pages/__fixtures__/teamMock'
import { HOME_PAGE_QUERY } from '../pages/Home'
import MockedQuery from '../utils/test-utils/MockedQuery'

const mocks = [
  {
    request: {
      query: HOME_PAGE_QUERY,
    },
    result: {
      data: {
        pageStart,
      },
    },
  },
  {
    request: {
      query: FOOTER_QUERY,
    },
    result: {
      data: {
        addresses,
      },
    },
  },
  TeamQueryMock,
]

describe('App', () => {
  it('renders App', async () => {
    const { container } = render(
      <MockedQuery location={{ pathname: '/' }} mocks={mocks}>
        <App />
      </MockedQuery>
    )

    await wait()

    expect(container).toMatchSnapshot()
  })
})
