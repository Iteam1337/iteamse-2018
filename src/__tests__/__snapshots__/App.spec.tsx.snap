// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
<div>
  <div
    class="sc-eLExRp hhMQTZ"
  >
    <button
      class="sc-exAgwC jqAmCX"
      aria-label="Menu button"
      data-testid="btn-mobile-menu"
      role="button"
    >
      <div
        class="sc-gojNiO ixAttR"
      >
        <div
          class="sc-cQFLBn clVStQ"
        />
        <div
          class="sc-cQFLBn clVStQ"
        />
        <div
          class="sc-cQFLBn clVStQ"
        />
      </div>
    </button>
  </div>
  <div
    class="sc-hMFtBS bNtZdn"
  >
    <div
      class="sc-cLQEGU HJkVy"
    >
      <div
        class="sc-hZSUBg fMTTZm"
      >
        <div
          class="sc-cMhqgX kHrnQp"
        >
          Iteam Göteborg
        </div>
        <div
          class="sc-iuJeZd glPZiI"
        >
          <ul
            class="sc-kAzzGY bxsBVD"
          >
            <li>
              Haga Kyrkogata 28
            </li>
            <li>
              411 23
              Göteborg
            </li>
          </ul>
        </div>
        <ul
          class="sc-kAzzGY bxsBVD"
        >
          <li>
            +46 734 029 112
          </li>
          <li
            class="sc-cmthru WOJnp"
          >
            <a
              class="sc-esOvli dMIGps"
            >
              johanna.grahn@iteam.se
            </a>
          </li>
          <li>
            ORG: 
            556551-6928
          </li>
        </ul>
      </div>
      <div
        class="sc-hZSUBg fMTTZm"
      >
        <div
          class="sc-cMhqgX kHrnQp"
        >
          Iteam Stockholm
        </div>
        <div
          class="sc-iuJeZd glPZiI"
        >
          <ul
            class="sc-kAzzGY bxsBVD"
          >
            <li>
              Östermalmsgatan 26A
            </li>
            <li>
              114 26
              Stockholm
            </li>
          </ul>
        </div>
        <ul
          class="sc-kAzzGY bxsBVD"
        >
          <li>
            +46 826 70 90
          </li>
          <li
            class="sc-cmthru WOJnp"
          >
            <a
              class="sc-esOvli dMIGps"
            >
              info@iteam.se
            </a>
          </li>
          <li>
            ORG: 
            556551-6928
          </li>
        </ul>
      </div>
      <div
        class="sc-hORach jBrJiT"
      >
        <div
          class="sc-gqPbQI nneVN"
        >
          Följ oss
        </div>
        <ul
          class="sc-kAzzGY bxsBVD"
        >
          <li
            class="sc-bMVAic dltDGj"
          >
            <img
              class="sc-bAeIUo ghSJa"
              src="test-file-stub"
              alt=""
            />
            <a
              class="sc-iujRgT kPAKfD"
              href="https://www.facebook.com/iteam1337/"
            >
              Facebook
            </a>
          </li>
          <li
            class="sc-bMVAic dltDGj"
          >
            <img
              class="sc-bAeIUo ghSJa"
              src="test-file-stub"
              alt=""
            />
            <a
              class="sc-iujRgT kPAKfD"
              href="https://instagram.com/iteam1337"
            >
              Instagram
            </a>
          </li>
          <li
            class="sc-bMVAic dltDGj"
          >
            <img
              class="sc-bAeIUo ghSJa"
              src="test-file-stub"
              alt=""
            />
            <a
              class="sc-iujRgT kPAKfD"
              href="https://www.linkedin.com/company/270569/"
            >
              LinkedIn
            </a>
          </li>
          <li
            class="sc-bMVAic dltDGj"
          >
            <img
              class="sc-bAeIUo ghSJa"
              src="test-file-stub"
              alt=""
            />
            <a
              class="sc-iujRgT kPAKfD"
              href="https://www.youtube.com/channel/UCU2TdLZ7p0jEuuGUxaod4lw"
            >
              YouTube
            </a>
          </li>
          <li
            class="sc-bMVAic dltDGj"
          >
            <img
              class="sc-bAeIUo ghSJa"
              src="test-file-stub"
              alt=""
            />
            <a
              class="sc-iujRgT kPAKfD"
              href="http://www.mynewsdesk.com/se/iteam"
            >
              MyNewsDesk
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="sc-GMQeP jFfOBi"
    >
      <a
        class="sc-jnlKLf ewlqyw"
        black="1"
      >
        Behöver du supporthjälp?
      </a>
    </div>
  </div>
</div>
`;
